src/WCSimWCAddDarkNoise.cc

```diff
@@ -0,0 +1,254 @@
+#include "WCSimWCAddDarkNoise.hh"
+#include "WCSimWCPMT.hh"
+#include "WCSimWCDigi.hh"
+#include "WCSimWCHit.hh"
+#include "G4EventManager.hh"
+#include "G4Event.hh"
+#include "G4SDManager.hh"
+#include "G4DigiManager.hh"
+#include "G4ios.hh"
+#include "G4RotationMatrix.hh"
+#include "G4ThreeVector.hh"
+#include "G4LogicalVolumeStore.hh"
+
+#include "WCSimDetectorConstruction.hh"
+#include "WCSimPmtInfo.hh"
+#include "WCSimDarkRateMessenger.hh"
+
+#include <vector>
+#include <utility>
+// for memset
+#include <cstring>
+#include <iostream>
+
+#ifndef WCSIMWCADDDARKNOISE_VERBOSE
+//#define WCSIMWCADDDARKNOISE_VERBOSE
+#endif
+
+WCSimWCAddDarkNoise::WCSimWCAddDarkNoise(G4String name,
+                                   WCSimDetectorConstruction* myDetector)
+  :G4VDigitizerModule(name)
+{
+  this->myDetector = myDetector;
+  DigiHitMap.clear();
```

Comment:
Looking into this a bit more, it seems that if both omittable and currentValueAsDefault are set to true, GetCurrentValue() should be called and SetNewValue() called afterwards with this information. A quick look at doing things this way was unsuccessful.
So I just set omittable to false as you suggested.